core/src/iosMain/kotlin/com/powersync/DatabaseDriverFactory.ios.kt

```diff
@@ -51,10 +53,13 @@ public actual class DatabaseDriverFactory {
         scope: CoroutineScope,
         dbFilename: String,
     ): PsSqlDriver {
+        val dbDispatcher = Dispatchers.IO
+        val dbScope = CoroutineScope(scope.coroutineContext + dbDispatcher)
```

Comment:
I'd be careful with using the `CoroutineScope` interface here. The docs mention `Manual implementation of this interface is not recommended, implementation by delegation should be preferred instead.` https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-coroutine-scope/

It might be better to use `coroutineScope`.
Some info from my GPT friend: `coroutineScope` is used for Structured concurrency. Structured concurrency guarantees:

- When a scope cancels, all of its coroutines cancel.
- When a suspend function returns, all of its work is done.
- When a coroutine errors, its caller or scope is notified.
The coroutineScope unlike the CoroutineScope, will suspend itself until all coroutines started inside of it are complete. Indeed it creates a local scope and guarantees that all child coroutines will be completed before the return instruction.
https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/coroutine-scope.html